src/ovs/aca_ovs_control.cpp

```diff
@@ -164,68 +200,73 @@ void ACA_OVS_Control::parse_packet(uint32_t in_port, void *packet)
       ACA_LOG_INFO("   Protocol: unknown\n");
     }
   }
-  /* define/compute tcp header offset */
-  const struct sniff_tcp *tcp =
-          (struct sniff_tcp *)(base + SIZE_ETHERNET + vlan_len + size_ip);
-  const unsigned char *payload;
-  int size_payload;
-  int size_tcp = TH_OFF(tcp) * 4;
-
-  if (size_tcp < 20) {
-    // printf("   *** Invalid TCP header length: %u bytes\n", size_tcp);
-    return;
-  } else {
-    ACA_LOG_INFO("   Src port: %d\n", ntohs(tcp->th_sport));
-    ACA_LOG_INFO("   Dst port: %d\n", ntohs(tcp->th_dport));
-
-    /* define/compute tcp payload (segment) offset */
-    payload = (u_char *)(base + SIZE_ETHERNET + vlan_len + size_ip + size_tcp);
 
-    /* compute tcp payload (segment) size */
-    size_payload = ntohs(ip->ip_len) - (size_ip + size_tcp);
-
-    /*
-        * Print payload data;
-        */
-    if (size_payload > 0) {
-      ACA_LOG_INFO("   Payload (%d bytes):\n", size_payload);
-      print_payload(payload, size_payload);
-    }
-  }
-
-  /* define/compute udp header offset */
-  const struct sniff_udp *udp =
-          (struct sniff_udp *)(base + SIZE_ETHERNET + vlan_len + size_ip);
-  int size_udp = ntohs(udp->uh_ulen);
-
-  if (size_udp < 20) {
-    ACA_LOG_ERROR("size_udp < 20: %d\n", size_udp);
-    return;
-  } else {
-    int udp_sport = ntohs(udp->uh_sport);
-    int udp_dport = ntohs(udp->uh_dport);
-    ACA_LOG_INFO("   Src port: %d\n", udp_sport);
-    ACA_LOG_INFO("   Dst port: %d\n", udp_dport);
-
-    /* define/compute udp payload (daragram) offset */
-    payload = (u_char *)(base + SIZE_ETHERNET + vlan_len + size_ip + 8);
-
-    /* compute udp payload (datagram) size */
-    size_payload = ntohs(ip->ip_len) - (size_ip + 8);
-
-    /* Print payload data. */
-    if (size_payload > 0) {
-      ACA_LOG_INFO("   Payload (%d bytes):\n", size_payload);
-      print_payload(payload, size_payload);
-    }
+  if (ip->ip_p == IPPROTO_TCP) {
+    /* define/compute tcp header offset */
+    const struct sniff_tcp *tcp =
+            (struct sniff_tcp *)(base + SIZE_ETHERNET + vlan_len + size_ip);
+    const unsigned char *payload;
+    int size_payload;
+    int size_tcp = TH_OFF(tcp) * 4;
+
+    if (size_tcp < 20) {
+      ACA_LOG_ERROR("size_tcp < 20: %d bypes \n", size_tcp);
+      return;
+    } else {
+      ACA_LOG_INFO("   Src port: %d\n", ntohs(tcp->th_sport));
+      ACA_LOG_INFO("   Dst port: %d\n", ntohs(tcp->th_dport));
+
+      /* define/compute tcp payload (segment) offset */
+      payload = (u_char *)(base + SIZE_ETHERNET + vlan_len + size_ip + size_tcp);
+
+      /* compute tcp payload (segment) size */
+      size_payload = ntohs(ip->ip_len) - (size_ip + size_tcp);
+
+      /*
+          * Print payload data;
+          */
+      if (size_payload > 0) {
+        ACA_LOG_INFO("   Payload (%d bytes):\n", size_payload);
+        print_payload(payload, size_payload);
```

Comment:
let's remove this line also to reduce the excessive syslog output.

Reply:
print_payload removed